### PR TITLE
Feature: DeveLanCacheUI service widget

### PR DESCRIPTION
**Pull request by devedse**
## Proposed change

Closes: https://github.com/gethomepage/homepage/discussions/3543

image

I already created a PR for this previously but couldn't get things to work back then. In this PR everything is tested locally and works fine.

Also added the icon in a PR to dashboard-icons:
https://github.com/walkxcode/dashboard-icons/pull/766

## Type of change

- [X] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [X] If applicable, I have added corresponding documentation changes.
- [X] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [X] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [X] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

**Comment by shamoon** [comment]
Thanks, however:

- Please read the guidelines for opening a PR, this can be reopened later.
- Also, you should only edit the English translation file

**Comment by devedse** [comment]
Ah, you mean the amount of upvotes on the feature request is top low? I remember it being 5. We'll wait for some more interest to boil up then.

**Comment by devedse** [comment]
@shamoon , I saw that I have since reached 12 upvotes. Can you give me a list of changes that need to happen in this PR?

Kind regards,
Devedse

**Comment by shamoon** [comment]
It’s very surprising that a project with a seemingly small audience managed to do that in 12 hours, don’t you think? And not a single comment. Any thoughts?

**Comment by devedse** [comment]
I asked in the discord of LanCache if people where interested. Apparently they were.

**Comment by shamoon** [comment]
I’ll review the PR when I have a sec, you can just remove the other lang file changes for now

File: docs/widgets/services/develancacheui.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Ensure you point this to the API and not the frontend. Example:
+
+```yaml
+- DeveLanCacheUI:
+    icon: /assets/icons/DeveLanCacheUI.png
+    href: https://develancacheui.devedse.duckdns.org
+    widget:
+        type: develancacheui
+        url: https://develancacheui_api.devedse.duckdns.org #Note _api here
```
Comment (shamoon) [inline docs/widgets/services/develancacheui.md:22]: What does this mean? Also, don't use a real URL.
Also fails lint it looks like

File: docs/widgets/services/develancacheui.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+widget:
+  type: develancacheui
+  url: http://your.fileflows.host:port
+```
+
+Ensure you point this to the API and not the frontend. Example:
+
+```yaml
+- DeveLanCacheUI:
+    icon: /assets/icons/DeveLanCacheUI.png
```
Comment (shamoon) [inline docs/widgets/services/develancacheui.md:18]:
```suggestion
```

File: docs/widgets/services/develancacheui.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    href: https://develancacheui.devedse.duckdns.org
```
Comment (shamoon) [inline docs/widgets/services/develancacheui.md:19]:
```suggestion
    href: https://develancacheui.host.or.ip
```

File: src/widgets/develancacheui/widget.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import genericProxyHandler from "utils/proxy/handlers/generic";
+
+const widget = {
+  api: "{url}/{endpoint}",
+  proxyHandler: genericProxyHandler,
+
+  mappings: {
+    "DownloadStats/GetTotalDownloadStats": {
```
Comment (shamoon) [inline src/widgets/develancacheui/widget.js:8]:
```suggestion
    "stats": {
```

File: src/widgets/develancacheui/component.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+export default function Component({ service }) {
+  const { t } = useTranslation();
+
+  const { widget } = service;
+
+  const { data: downloadStatsData, error: downloadStatsError } = useWidgetAPI(
+    widget,
+    "DownloadStats/GetTotalDownloadStats",
+  );
```
Comment (shamoon) [inline src/widgets/develancacheui/component.jsx:15]:
```suggestion
  const { data: downloadStatsData, error: downloadStatsError } = useWidgetAPI(widget, "stats");
```

File: docs/widgets/services/develancacheui.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+description: DeveLanCacheUI Widget Configuration
+---
+
+Learn more about [DeveLanCacheUI](https://github.com/devedse/DeveLanCacheUI_Backend).
+
+```yaml
+widget:
+  type: develancacheui
+  url: http://your.fileflows.host:port
+```
```
Comment (shamoon) [inline docs/widgets/services/develancacheui.md:12]:
```suggestion
```

copypasta I assume

**Review by shamoon — CHANGES_REQUESTED** [review]

File: docs/widgets/services/develancacheui.md (hunk 1 above)
Comment (devedse) [reply docs/widgets/services/develancacheui.md]: Rewrote the .md file

File: src/widgets/develancacheui/component.jsx (hunk 5 above)
Comment (devedse) [reply src/widgets/develancacheui/component.jsx]: Fixed

File: src/widgets/develancacheui/widget.js (hunk 4 above)
Comment (devedse) [reply src/widgets/develancacheui/widget.js]: fixed

**Comment by devedse** [comment]
I think I've resolved all your comments. Feel free to re-review and let me know if there's anything else to be improved.

**Review by shamoon — APPROVED** [review]

**Pull request merged**